### PR TITLE
Support serializing and deserializing null in Avro serdes

### PR DESCRIPTION
**Pull request by YairHalberstadt**
Fixes https://github.com/confluentinc/confluent-kafka-dotnet/issues/905

Brings this up to par with the java implementation.
One limitation is that you cannot produce or consume nulls of value type primitives.

### CODE REVIEW

**Comment by timtebeek** [comment]
@mhowlett Could you weigh in here? Would love to see this merged & released for our landscape of mixed Java/.Net services, so we can use the same patterns of tombstone records throughout.

**Comment by kuduh** [comment]
@mhowlett When will the error of serialization and deserialization of null in Avro serdes be fixed?

File: src/Confluent.SchemaRegistry.Serdes.Avro/AvroDeserializer.cs [hunk 1]
```diff
@@ -92,6 +92,19 @@ public async Task<T> DeserializeAsync(ReadOnlyMemory<byte> data, bool isNull, Se
         {
             try
             {
+                if (isNull)
+                {
+                    // Jit optimizes this check away
+                    if (default(T) == null)
```
Comment (mhowlett) [inline src/Confluent.SchemaRegistry.Serdes.Avro/AvroDeserializer.cs:98]: nice

**Comment by mhowlett** [comment]
Thanks @YairHalberstadt - sorry for the delay. This is looking good to me - the trick you're playing with `default` is something I think I didn't think of when I went to do this. I want to spend a bit of time playing with it to make sure though before merging. Targeting v1.9.0.

**Review by mhowlett — APPROVED** [review]
Great work, thanks for doing this. It'll be v1.9.0

**Comment by ahmar-husain** [comment]
@mhowlett when v.1.9.0 will be available?

**Pull request merged**